src/ripple/protocol/AcctRoot.h

```diff
@@ -0,0 +1,377 @@
+    AcctRootImpl(AcctRootImpl&&) = default;
+
+    AcctRootImpl&
+    operator=(AcctRootImpl const& rhs)
+    {
+        Base::operator=(rhs);
+        return *this;
+    }
+
+    AcctRootImpl&
+    operator=(AcctRootImpl&& rhs)
+    {
+        Base::operator=(std::move(rhs));
+        return *this;
+    }
+
+    ~AcctRootImpl() = default;
+
+    // Conversion operator from AcctRootImpl<true> to AcctRootImpl<false>.
+    operator AcctRootImpl<true>() const
+    {
+        return AcctRootImpl<false>(
+            std::const_pointer_cast<std::shared_ptr<STLedgerEntry const>>(
+                wrapped_));
+    }
+
+    [[nodiscard]] AccountID
+    accountID() const
+    {
+        return wrapped_->at(sfAccount);
+    }
+
+    [[nodiscard]] std::uint32_t
+    sequence() const
+    {
+        return wrapped_->at(sfSequence);
+    }
+
+    void
+    setSequence(std::uint32_t seq)
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        wrapped_->at(sfSequence) = seq;
+    }
+
+    [[nodiscard]] STAmount
+    balance() const
+    {
+        return wrapped_->at(sfBalance);
+    }
+
+    void
+    setBalance(STAmount const& amount)
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        wrapped_->at(sfBalance) = amount;
+    }
+
+    [[nodiscard]] std::uint32_t
+    ownerCount() const
+    {
+        return wrapped_->at(sfOwnerCount);
+    }
+
+    void
+    setOwnerCount(std::uint32_t newCount)
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        wrapped_->at(sfOwnerCount) = newCount;
+    }
+
+    [[nodiscard]] std::uint32_t
+    previousTxnID() const
+    {
+        return wrapped_->at(sfOwnerCount);
+    }
+
+    void
+    setPreviousTxnID(uint256 prevTxID)
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        wrapped_->at(sfPreviousTxnID) = prevTxID;
+    }
+
+    [[nodiscard]] std::uint32_t
+    previousTxnLgrSeq() const
+    {
+        return wrapped_->at(sfPreviousTxnLgrSeq);
+    }
+
+    void
+    setPreviousTxnLgrSeq(std::uint32_t prevTxLgrSeq)
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        wrapped_->at(sfPreviousTxnLgrSeq) = prevTxLgrSeq;
+    }
+
+    [[nodiscard]] std::optional<uint256>
+    accountTxnID() const
+    {
+        return wrapped_->at(~sfAccountTxnID);
+    }
+
+    void
+    setAccountTxnID(uint256 const& newAcctTxnID)
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        this->setOptional(sfAccountTxnID, newAcctTxnID);
+    }
+
+    void
+    clearAccountTxnID()
+    {
+        static_assert(Writable, "Cannot set member of const ledger entry.");
+        this->clearOptional(sfAccountTxnID);
+    }
+
+    [[nodiscard]] std::optional<AccountID>
+    regularKey() const
+    {
+        return wrapped_->at(~sfRegularKey);
+    }
+
+    void
+    setRegularKey(AccountID const& newRegKey)
```

Comment:
Can we add a `requires Writable` and lose the `static_assert` (here and all the other non-const functions)? I think this is cleaner:
```
    void
    setRegularKey(AccountID const& newRegKey)
        requires Writable
    {
        this->setOptional(sfRegularKey, newRegKey);
    }
```

Reply:
I just noticed the "this". I don't really object, but we usually don't put "this" in front of member functions. Is there a reason it's done here?